### ISSUE TITLE
Break free from ECMAScript

### ISSUE THREAD

**Issue by FeldrinH** [TodePond/DreamBerd#234]
Based on the discussion in https://github.com/TodePond/DreamBerd/issues/229 I present a new proposal:

As per https://github.com/TodePond/DreamBerd/issues/229, DreamBerd is currently equivalent to ECMAScript unless stated otherwise. I propose getting rid of this dependence on ECMAScript and making the DreamBerd specification self-contained.

This would have the following benefits:

As a conceptual benefit, it would avoid any of the imperfections of ECMAScript leaking into DreamBerd, thus ensuring DreamBerd is truly perfect.

As a practical benefit, this would make implementing a fully compliant DreamBerd interpreter much more doable. Currently, even if you are willing to integrate an ECMAScript runtime into your interpreter, you still have to somehow make DreamBerd's `when`, `previous`, `next`, `async` turns, mutability restrictions, lifetimes and whatnot work together with all the ECMAScript features (which are extremely numerous).

**Comment by interrealm** [comment]
YES

**Comment by Bytestorm5** [comment]
Personally I think having it in ECMAScript is funnier 

I do also know that most of what's in the specification is actually possible in ECMAScript, so it should be possible to write a JS or TS interpreter.

**Comment by FeldrinH** [comment]
> Personally I think having it in ECMAScript is funnier
> 
> I do also know that most of what's in the specification is actually possible in ECMAScript, so it should be possible to write a JS or TS interpreter.

I disagree, but I will watch your efforts to implement an interpreter with great interest.
